### ISSUE TITLE
Easy Acceptor Error Templates Don't Work

### ISSUE THREAD

**Issue by spyrosoft** [edicl/hunchentoot#112]
I'm probably doing it wrong. Here is the most minimal example I can think of:
```common-lisp
(hunchentoot:start (make-instance 'hunchentoot:easy-acceptor :document-root "." :error-template-directory "error-templates" :port 8080))
```

I expect the error-templates/404.html file to display when I visit http://localhost:8080. Thoughts?

**Comment by hanshuebner** [comment]
':document-root` and `:error-template-directory` accept a pathname, and a directory pathname ends with a slash. Try passing *default-pathname-defaults* and "error-templates/" instead.

**Comment by stassats** [comment]
Maye it should call cl-fad:pathname-as-directory on it?

**Comment by hanshuebner** [comment]
If you like.  I'm more thinking that pathname syntax is what it is, and if one does not understand it, surprises will ensue at some point anyway.  Rather than silently coercing, maybe warning about a non-existent directory would be more helpful?  The warning could mention that the slash at the end is missing, if that is the case.

**Comment by stassats** [comment]
I don't think it's hunchentoot's job to teach people about pathnames, and if it means one less github ticket then it would be worth it.

**Comment by hanshuebner** [comment]
Go for it.

2015-12-10 10:07 GMT+01:00 Stas Boukarev <notifications@github.com>:

**Comment by stassats** [comment]
Ok, I will, a bit later. This also could be viewed as a safety feature, since one might inadvertently expose a wrong directory.

**Comment by spyrosoft** [comment]
I would like to restate that I'm sure what's wrong is my not understanding how to do this properly. However, no matter what I try I can't get the error-template-directory to work. I tried pathnames as well prior to posting this issue, I was just trying to boil down the code as much as I could to avoid clutter. Here is a more detailed example:

```common-lisp
(defvar *hunchentoot-directory*
  (pathname (directory-namestring #.(or *compile-file-pathname* *load-truename*))))

(defvar example-website-server
  (make-instance 'hunchentoot:easy-acceptor
                 :document-root *hunchentoot-directory*
                 :error-template-directory (merge-pathnames "static/error-templates/" *hunchentoot-directory*)
                 :access-log-destination (merge-pathnames "logs/access.log" *hunchentoot-directory*)
                 :message-log-destination (merge-pathnames "logs/message.log" *hunchentoot-directory*)
                 :port 8480))
```

When I print (merge-pathnames "static/error-templates/" *hunchentoot-directory*), it gives me #P"/home/username/hunchentoot/static/error-templates/". 

The file exists and the permissions are fine.
```shell
ls /home/spyrosoft/hunchentoot/static/error-templates/ 
404.html
```

I'm stumped.

**Comment by stassats** [comment]
You have to set *show-lisp-errors-p* to T, which was changed in
https://github.com/edicl/hunchentoot/commit/610b535760431efecc71e850d5834638d196b81a
never documented, and I doubt is a good idea overall.

**Comment by spyrosoft** [comment]
Yes! It's working! Thank you, stassats. Thoughts on leaving it, but making the default T?

**Comment by stassats** [comment]
Not showing lisp errors and showing error templates should be separate concerns. It got disabled because the default 500 template discloses the error and backtrace, which may be unsafe.
Maybe Hans or Edi have a suggestion.
